app/plugin_api.go

```diff
@@ -323,6 +323,23 @@ func (api *PluginAPI) GetFileInfo(fileId string) (*model.FileInfo, *model.AppErr
 	return api.app.GetFileInfo(fileId)
 }
 
+func (api *PluginAPI) GetFileLink(fileId string) (string, *model.AppError) {
+	if !api.app.Config().FileSettings.EnablePublicLink {
+		return "", model.NewAppError("GetFileLink", "plugin_api.get_file_link.disabled.app_error", nil, "", http.StatusNotImplemented)
```

Comment:
@jwilander, do we need an error message to accompany the new `plugin_api.get_file_link.disabled.app_error` and `plugin_api.get_file_link.no_post.app_error`? I can't quite recall our stance on translating these kind of error messages.

Reply:
Ideally we'd have an error message that's __not__ translated but I'm not sure AppError supports that at the moment. I think this falls under general improvements to AppError in the future

Reply:
Makes sense. For now, @Charliekenney23 can you update our `i18n/en.json` to add messages for these error ids?